### PR TITLE
Post-PR cleanups

### PR DESCRIPTION
**Pull request by TomasPuverle**
Addressing #1828.

This is ready to go but @b-scholz mentioned that he would prefer to merge this after the RAM changes are merged.

### CODE REVIEW

**Review by b-scholz — APPROVED** [review]

File: src/synthesiser/Synthesiser.cpp [hunk 1]
```diff
@@ -1961,7 +1961,7 @@ void Synthesiser::emitCode(std::ostream& out, const Statement& stmt) {
 #define UNARY_OP(opcode, ty, op)                \
     case FunctorOp::opcode: {                   \
         out << "(" #op "(ramBitCast<" #ty ">("; \
-        visit(*args[0], out);                    \
+        dispatch(*args[0], out);                    \
```
Comment (b-scholz) [inline src/synthesiser/Synthesiser.cpp:1964]: This is picky but can you align the backslash. @mmcgr would insist on this.

File: src/synthesiser/Synthesiser.cpp [hunk 2]
```diff
@@ -1976,9 +1976,9 @@ void Synthesiser::emitCode(std::ostream& out, const Statement& stmt) {
 #define BINARY_OP_EXPR_EX(ty, op, rhs_post)      \
     {                                            \
         out << "(ramBitCast<" #ty ">(";          \
-        visit(*args[0], out);                     \
+        dispatch(*args[0], out);                     \
```
Comment (b-scholz) [inline src/synthesiser/Synthesiser.cpp:1979]: Here again.

File: src/synthesiser/Synthesiser.cpp [hunk 3]
```diff
@@ -1976,9 +1976,9 @@ void Synthesiser::emitCode(std::ostream& out, const Statement& stmt) {
...
         out << ") " #op " ramBitCast<" #ty ">("; \
-        visit(*args[1], out);                     \
+        dispatch(*args[1], out);                     \
```
Comment (b-scholz) [inline src/synthesiser/Synthesiser.cpp:1981]: and here

File: src/synthesiser/Synthesiser.cpp [hunk 4]
```diff
@@ -2005,9 +2005,9 @@ void Synthesiser::emitCode(std::ostream& out, const Statement& stmt) {
 #define BINARY_OP_EXP(opcode, ty, tyTemp)                                                     \
     case FunctorOp::opcode: {                                                                 \
         out << "static_cast<" #ty ">(static_cast<" #tyTemp ">(std::pow(ramBitCast<" #ty ">("; \
-        visit(*args[0], out);                                                                  \
+        dispatch(*args[0], out);                                                                  \
```
Comment (b-scholz) [inline src/synthesiser/Synthesiser.cpp:2008]: and here

File: src/synthesiser/Synthesiser.cpp [hunk 5]
```diff
@@ -2005,9 +2005,9 @@ void Synthesiser::emitCode(std::ostream& out, const Statement& stmt) {
...
         out << "), ramBitCast<" #ty ">(";                                                     \
-        visit(*args[1], out);                                                                  \
+        dispatch(*args[1], out);                                                                  \
```
Comment (b-scholz) [inline src/synthesiser/Synthesiser.cpp:2010]: and here

File: src/synthesiser/Synthesiser.cpp [hunk 6]
```diff
@@ -2017,7 +2017,7 @@ void Synthesiser::emitCode(std::ostream& out, const Statement& stmt) {
         out << #op "({";                   \
         for (auto& cur : args) {           \
             out << "ramBitCast<" #ty ">("; \
-            visit(*cur, out);               \
+            dispatch(*cur, out);               \
```
Comment (b-scholz) [inline src/synthesiser/Synthesiser.cpp:2020]: and here

File: src/synthesiser/Synthesiser.cpp [hunk 7]
```diff
@@ -2032,13 +2032,13 @@ void Synthesiser::emitCode(std::ostream& out, const Statement& stmt) {
 #define CONV_TO_STRING(opcode, ty)                \
     case FunctorOp::opcode: {                     \
         out << "symTable.lookup(std::to_string("; \
-        visit(*args[0], out);                      \
+        dispatch(*args[0], out);                      \
```
Comment (b-scholz) [inline src/synthesiser/Synthesiser.cpp:2035]: and here

File: src/synthesiser/Synthesiser.cpp [hunk 8]
```diff
@@ -2032,13 +2032,13 @@ void Synthesiser::emitCode(std::ostream& out, const Statement& stmt) {
...
         out << "))";                              \
     } break;
 #define CONV_FROM_STRING(opcode, ty)                                            \
     case FunctorOp::opcode: {                                                   \
         out << "souffle::evaluator::symbol2numeric<" #ty ">(symTable.resolve("; \
-        visit(*args[0], out);                                                    \
+        dispatch(*args[0], out);                                                    \
```
Comment (b-scholz) [inline src/synthesiser/Synthesiser.cpp:2041]: and here

File: src/synthesiser/Synthesiser.cpp [hunk 9]
```diff
@@ -1712,7 +1712,7 @@ void Synthesiser::emitCode(std::ostream& out, const Statement& stmt) {
             // clang-format off
 #define EVAL_CHILD(ty, idx)        \
     out << "ramBitCast<" #ty ">("; \
-    visit(rel.idx(), out);         \
+    dispatch(rel.idx(), out);         \
```
Comment (b-scholz) [inline src/synthesiser/Synthesiser.cpp:1715]: Please can you align the backslash.

**Review by b-scholz — COMMENTED** [review]
Thanks for the contribution!

File: src/synthesiser/Synthesiser.cpp (hunk 1 above)
Comment (TomasPuverle) [reply src/synthesiser/Synthesiser.cpp:1964]: Done

File: src/synthesiser/Synthesiser.cpp (hunk 3 above)
Comment (TomasPuverle) [reply src/synthesiser/Synthesiser.cpp:1981]: Done

File: src/synthesiser/Synthesiser.cpp (hunk 5 above)
Comment (TomasPuverle) [reply src/synthesiser/Synthesiser.cpp:2010]: Done

File: src/synthesiser/Synthesiser.cpp (hunk 6 above)
Comment (TomasPuverle) [reply src/synthesiser/Synthesiser.cpp:2020]: Done

File: src/synthesiser/Synthesiser.cpp (hunk 7 above)
Comment (TomasPuverle) [reply src/synthesiser/Synthesiser.cpp:2035]: Done.

File: src/synthesiser/Synthesiser.cpp (hunk 8 above)
Comment (TomasPuverle) [reply src/synthesiser/Synthesiser.cpp:2041]: Done

File: src/synthesiser/Synthesiser.cpp (hunk 9 above)
Comment (TomasPuverle) [reply src/synthesiser/Synthesiser.cpp:1715]: Done

File: src/synthesiser/Synthesiser.cpp (hunk 4 above)
Comment (TomasPuverle) [reply src/synthesiser/Synthesiser.cpp:2008]: Done

File: src/synthesiser/Synthesiser.cpp (hunk 2 above)
Comment (TomasPuverle) [reply src/synthesiser/Synthesiser.cpp:1979]: Done

**Review by b-scholz — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**